src/libclipper/src/selection_policies.cpp

```diff
@@ -60,6 +61,7 @@ std::vector<PredictTask> DefaultOutputSelectionPolicy::select_predict_tasks(
     log_error_formatted(LOGGING_TAG_SELECTION_POLICY,
                         "No candidate models for query with label {}",
                         query.label_);
+    throw NoModelsFoundError();
```

Comment:
Thanks for the feedback. Removed the exception and added a check against `task_futures.empty()` instead.